src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -89,7 +89,11 @@ public virtual Stream GetNuspec()
         {
             // This is the default implementation. It is overridden and optimized in
             // PackageArchiveReader and PackageFolderReader.
+            return GetStream(GetNuspecFile());
+        }
 
+        public virtual string GetNuspecFile()
+        {
             // Find all nuspecs in the root folder.
             var nuspecPaths = GetFiles()
                 .Where(entryPath => PackageHelper.IsManifest(entryPath))
```

Comment:
Wouldn't it make sense to always do :
f => f.TrimStart('/').Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar)   ???